api/krusty/kustomizer.go

```diff
@@ -89,8 +90,18 @@ func (b *Kustomizer) Run(
 	if err != nil {
 		return nil, err
 	}
-	if b.options.DoLegacyResourceSort {
-		err = builtins.NewLegacyOrderTransformerPlugin().Transform(m)
+	if b.options.DoLegacyResourceSort &&
```

Comment:
Ah I see. I think as long as the flag's behavior is currently backwards compatible (meaning the behavior doesn't change if the flag is set and there is no sort order in the kustomization file), I don't think it matters too much what we do in the scenario that the flag is set and the sort order field is specified. 

After this PR goes in we should deprecate the `--reorder` flag with a message to use this field in the kustomization filed instead of the flag, so that it is clear to users that they're not intended to be used together.

Reply:
Actually, I've thought about it a bit more and I think since we are replacing the `--reorder` flag with this `sortOptions` field I'm inclined to think that we should throw an error if both are set. WDYT?